### ISSUE TITLE
Failed cluster install causes clusterdeployment unbound PersistentVolumeClaims 

### ISSUE THREAD

**Issue by chrisahl** [openshift/hive#828]
OCP 4.2 (1 master, 3 workers) with MCM and hive (v1-alpha-2020-02-12)

ClusterDeployment shows this error:

```
Events:
  Type     Reason            Age                 From               Message
  ----     ------            ----                ----               -------
  Warning  FailedScheduling  74s (x31 over 37m)  default-scheduler  pod has unbound immediate PersistentVolumeClaims 
```

In slack, @dgoodwin remarked:
```his is for a feature where we attempt to gather logs from a failed cluster install, we assumed any cluster where hive would run would have storage configured

i'm guessing yours does not

we've seen this elsewhere as well

so to workaround oc edit hiveconfig hive

and set
spec.failedProvisionConfig.skipGatherLogs = true

we have a card to make this the default so you opt into it instead of out
```

For now, MCM will add the following to the install scripting:
```
oc patch hiveconfig hive  --type json -p '[{"op": "replace", "path":"/spec/failedProvisionConfig/skipGatherLogs", "value":true}]'

```

FYI @itdove @subbarao-meduri

**Comment by chrisahl** [comment]
@dgoodwin Any idea when the hive default will be spec.failedProvisionConfig.skipGatherLogs = true ?

**Comment by dgoodwin** [comment]
Not scheduled for this sprint and marked as low priority right now, should we bump?

**Comment by chrisahl** [comment]
@dgoodwin If skipGatherLogs=false will remain the default, can you give me more info on how to get storage setup for hive to use?

**Comment by staebler** [comment]
Is the install scripting to set the flag to true not working out as a solution?

**Comment by chrisahl** [comment]
In the older MCM, we were able to set the flag skipGatherLogs = true.  We are now switching to using an operator based install and could most likely set this, but we are concerned that we will really need mustGather logs for failed deployment debug.    Hence, we would need details on what needs to be done with hive for persistent storage.  Is there an example of hive being provisioned with persistent storage?  What are the min/avg/max size needed? 

Does the hive team find that the mustGather logs are needed always/sometimes/almost never?

**Comment by dgoodwin** [comment]
Relevant docs here: https://docs.openshift.com/container-platform/4.3/storage/understanding-persistent-storage.html

Nothing to do with hive per se it's just whether or not your cluster is configured with a default storage class I believe.

I think this feature of hive would qualify as "almost never" needed, but in rare cases it may pay off. I think I've heard of them being useful once. It often fails as it's quite flaky. Most recently we needed them and they weren't there because someone was using bad network config and for whatever reason, that meant failed cluster log gathering.

**Comment by gregsheremeta** [comment]
would be fixed by #1150
